From 59b431a1b545c2ede92a983ae434166390bc218b Mon Sep 17 00:00:00 2001
From: Eric Chanudet <chanudete@ainfosec.com>
Date: Mon, 1 Jul 2019 15:47:50 -0400
Subject: [PATCH] [efi] Fix sanity check in dbg_efi_protocols.

A null handle would have efi_handle_name() return NULL all the time,
might as well clarify the debug message.

Signed-off-by: Eric Chanudet <chanudete@ainfosec.com>
---
 src/interface/efi/efi_debug.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/interface/efi/efi_debug.c b/src/interface/efi/efi_debug.c
index 8ea0a822..9decb8c8 100644
--- a/src/interface/efi/efi_debug.c
+++ b/src/interface/efi/efi_debug.c
@@ -331,8 +331,7 @@ void dbg_efi_protocols ( EFI_HANDLE handle ) {
 
 	/* Sanity check */
 	if ( ! handle ) {
-		printf ( "HANDLE %s could not retrieve protocols\n",
-			 efi_handle_name ( handle ) );
+		printf ( "HANDLE is NULL, no protocol can be retrieved.\n" );
 		return;
 	}
 
-- 
2.22.0

